lib/matrix/matrix.rb

```diff
@@ -57,12 +61,54 @@ def self.convert(matrix)
     #
     #   Matrix[ [1,2], [3,4] ].each { |e| puts e }
     #     # => prints the numbers 1 to 4
-    def each(&block)
-      raise NotSupportedError unless block_given?
-
-      each_with_index { |elem, _, _| block.call(elem) }
-      self
-    end
+     def each(which = :all)
+        return to_enum :each, which unless block_given?
+        last = column_count - 1
+        case which
+        when :all
+          block = Proc.new
+          rows.each do |row|
```

Comment:
`rows` is slow, becouse make a copy. I think we need a simple cycle here.
@SID37, what do you think of it?

Reply:
I agree